### ISSUE TITLE
too long mempool chain error and 25 unconfirmed tx limit

### ISSUE THREAD

**Issue by ryanxcharles** [bitcoin-sv/bitcoin-sv#53]
The issue is that users cannot broadcast more than 25 unconfirmed transactions in a row. This is quickly becoming one of the largest issues with Money Button because people are trying to upload lots of files or multipart files to the blockchain. We're going to have to create some hacky workaround code.

We know you guys are already aware of the issue, but there wasn't already an issue on the github, and this would be a good place to help put information about it. It is definitely one of the top issues that affects people all the time. Seeing how often this issue affects people can help prioritize solving it.

The error we get is "too-long-mempool-chain".

Here is a conversation I had with Attila recently where what at first looked like a different bug turned out to be this issue:

<img width="920" alt="screen shot 2019-02-11 at 7 56 00 am" src="https://user-images.githubusercontent.com/63740/52575482-22667480-2dd3-11e9-9781-61491c7311a5.png">
<img width="904" alt="screen shot 2019-02-11 at 7 56 08 am" src="https://user-images.githubusercontent.com/63740/52575488-24303800-2dd3-11e9-80dc-280fbe5dbe0d.png">

**Comment by breign** [comment]
maybe dropping unconfirmed tx-es from mempool set by some timeout on local node would be nice too. 
P.S. it worx if you have like [key1,key2,...] ready, you just use new key.

**Comment by ravibhojwani** [comment]
Could be useful if Sindhiwallet gets “approved “

**Comment by ravibhojwani** [comment]
Thinking of building a brave browser extension- also...

**Comment by wildnothing** [comment]
Yes, this is currently blocking scalability of apps on BSV and needs to be addressed as a matter of priority.

We have a workaround involving kafka (partitioning by our users) and an exponential back-off retry policy in the case that a user reaches their limit per block. This scales horizontally but will become very, very expensive as our number of users grow.

We should not need to store our own pending blockchain operations off-chain. It defeats the point entirely and adds an unnecessary layer of middleware for blockchain fault tolerance. This is an oxymoron.

**Comment by Danconnolly** [comment]
The unconfirmed tx limit was increased to 10,000 in version 1.0.8.
